src/test/java/uk/gov/hmcts/reform/managecase/api/controller/CaseAssignmentControllerTest.java

```diff
@@ -172,4 +172,14 @@ void shouldFailWithBadRequestWhenCaseIdsInGetAssignmentsIsNull() throws Exceptio
         this.mockMvc.perform(get(CASE_ASSIGNMENTS))
                 .andExpect(status().isBadRequest());
     }
+
+    @DisplayName("should fail with 400 bad request when caseIds is malformed or invalid")
+    @Test
+    void shouldFailWithBadRequestWhenCaseIdsInGetAssignmentsIsMalformed() throws Exception {
+
+        this.mockMvc.perform(get(CASE_ASSIGNMENTS)
+                .queryParam("case_ids", "121324,%12345"))
+                .andExpect(status().isBadRequest())
+                .andExpect(jsonPath("$.message", is("Case ID should contain digits only")));
+    }
 }
```

Comment:
Data validation logic should be unit tested fully. Otherwise, the white-box testing will be weak, and compensating it with additional data validation scenarios will be an expensive activity.

Reply:
Now it has unit tests for all 4 scenarios.